### PR TITLE
Affiliates and affiliate users

### PR DESCRIPTION
**Pull request by iamanikeev**

### CODE REVIEW

File: tests/models/test_affiliate.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# coding: utf-8
+import pytest
+
+from pyoffers.exceptions import HasOffersException
+from pyoffers.models import Affiliate, AffiliateUser
+
+CASSETTE_NAME = 'affiliate'
```
Comment (Stranger6667) [inline tests/models/test_affiliate.py]: Please, add extra empty line above

Comment (iamanikeev) [reply tests/models/test_affiliate.py]: Done

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
Please, rebase and add documentation notes for the new model

**Comment by Stranger6667** [comment]
Also, fix isort build, which will fail I suppose (because of a single empty line after the imports in affiliate.py)

File: pyoffers/models/affiliate.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class Affiliate(Model):
+    """
+    An Affiliate.
+    """
+    generic_methods = ('update', 'delete')
+
+    @property
+    def affiliate_user(self):
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: Please, rename ``affiliate_user`` -> ``user``. I think, that this repetition will only decrease readability

File: pyoffers/models/affiliate.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        return RelatedUserManager(api=self._manager.api, base_manager_class=AffiliateUserManager, id=self.id).get()
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: AffiliateUser should be initialized here without extra code in ``__getattribute__`` and RelatedManager

File: pyoffers/models/core.py [hunk 4]
```diff
@@ -29,6 +29,18 @@ def generic_method(method):
     return method
 
 
+class RelatedManager:
```
Comment (Stranger6667) [inline pyoffers/models/core.py]: Could you, please, move this class to its initial place?

File: pyoffers/models/affiliate.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        'create',
+        'update',
+        'delete',
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+    )
+
+
+class RelatedUserManager(RelatedManager):
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: Probably, this class is redundant.

File: pyoffers/models/affiliate.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    name = 'affiliates'
+    generic_methods = (
+        'update',
+        'delete',
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+    )
+
+    def create(self, create_user=True, **kwargs):
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: Let's simplify the interface and remove creating of AffiliateUser. It still will be possible via ``api.affiliate_users.create``

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
I've added my comments on the changes

File: pyoffers/models/affiliate.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        'update',
+        'delete',
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+    )
+
+    def create(self, **kwargs):
+        """
+        Creates an affiliate and corresponding affiliate user (if ``create_user`` is set)
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: Docstring is obsolete

File: pyoffers/models/affiliate.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        :param kwargs:
+        :return: affiliate instance
+        """
+        return self._call('create', data=kwargs)
+
+    def create_with_user(self, user_params, **kwargs):
+        """
+        Creates an affiliate and corresponding affiliate user (if ``create_user`` is set)
```
Comment (Stranger6667) [inline pyoffers/models/affiliate.py]: This docstring is obsolete as well

File: pyoffers/models/core.py [hunk 9]
```diff
@@ -114,4 +122,4 @@ def __init__(self, api, base_manager_class, id):
 
     def find_all(self, **kwargs):
         kwargs[self.related_object_name] = self.id
-        return self.base_manager.find_all(**kwargs)
+        return self.base_manager.find_all(**kwargs)
```
Comment (Stranger6667) [inline pyoffers/models/core.py]: Missing newline

**Review by Stranger6667 — APPROVED** [review]

**Pull request merged**